main/src/main/python/embedding_generator/main.py

```diff
@@ -0,0 +1,119 @@
+from transformers import DebertaModel, AutoTokenizer
+import torch
+import numpy as np
+from datasets import load_dataset
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--model_type', dest='model_type', type=str)
+parser.add_argument('--tokenizer', dest='tokenizer', type=str)
+parser.add_argument('--context_size', dest='context_size', type=int)
+parser.add_argument('--device', dest='device', type=str)
+parser.add_argument('--huggingface_dataset', dest='huggingface_dataset', type=str)
+parser.add_argument('--huggingface_subpart', dest='huggingface_subpart', type=str)
+parser.add_argument('--number_of_tokens', dest='number_of_tokens', type=int)
+parser.add_argument('--output', dest='output', type=str)
+args = parser.parse_args()
+
+context = int(args.context_size) # 512
+model_type = args.model_type # microsoft/deberta-base
+tokenizer = args.tokenizer # microsoft/deberta-base
+device = args.device # cuda
+huggingface_dataset = args.huggingface_dataset # wikipedia
+huggingface_subpart = args.huggingface_subpart # wikipedia
+number_of_tokens = int(args.number_of_tokens) # 10000000
+output = str(args.output) # deberta_embd
+
+torch.set_default_device(device)
+
+model = DebertaModel.from_pretrained(model_type).to(device)
+tokenizer = AutoTokenizer.from_pretrained(tokenizer)
+
+if huggingface_dataset:
+    if huggingface_subpart:
+        dataset = load_dataset(huggingface_dataset, huggingface_subpart).with_format(type="torch")
+    else:
+        dataset = load_dataset(huggingface_dataset).with_format(type="torch")
+else:
+    dataset = load_dataset('json', data_files={'train': 'text_data.json'})
+
+tokens_so_far = 0
+crr = 0
+
+sum_tensors = {}
+crr_tensors = {}
+
+i = 0
+
+print(len(dataset))
+
+while tokens_so_far < number_of_tokens:  # 10 million
```

Comment:
Is this condition here because the input will likely have too many tokens to do them all once?  With the small test set, it means that they will all be done multiple times, perhaps unnecessarily.  The RNG might cause some indices to be visited multiple time and some none.  Could a slice of the data be used instead of the random samples with replacement?

Reply:
For reasonable data sets, the probability of repeating a token is exceedingly small. Considering that I'd say we need at least 10M tokens for a decent coverage. With a context size of 1024 that would mean we only look at roughly 10000 indices from our 10M total indices. The actual tokens might repeat but they will most likely be part of different contexts, thus having different embeddings.